src/pages/NewChatPage.js

```diff
@@ -44,226 +44,187 @@ const defaultProps = {
     reports: {},
 };
 
-class NewChatPage extends Component {
-    constructor(props) {
-        super(props);
-
-        this.toggleOption = this.toggleOption.bind(this);
-        this.createChat = this.createChat.bind(this);
-        this.createGroup = this.createGroup.bind(this);
-        this.updateOptionsWithSearchTerm = this.updateOptionsWithSearchTerm.bind(this);
-        this.excludedGroupEmails = _.without(CONST.EXPENSIFY_EMAILS, CONST.EMAIL.CONCIERGE);
-
-        const {recentReports, personalDetails, userToInvite} = OptionsListUtils.getNewChatOptions(
-            props.reports,
-            props.personalDetails,
-            props.betas,
-            '',
-            [],
-            this.props.isGroupChat ? this.excludedGroupEmails : [],
-        );
-        this.state = {
-            searchTerm: '',
-            recentReports,
-            personalDetails,
-            selectedOptions: [],
-            userToInvite,
-        };
-    }
-
-    componentDidUpdate(prevProps) {
-        if (_.isEqual(prevProps.reports, this.props.reports) && _.isEqual(prevProps.personalDetails, this.props.personalDetails)) {
-            return;
-        }
-        this.updateOptionsWithSearchTerm(this.state.searchTerm);
-    }
-
-    /**
-     * Returns the sections needed for the OptionsSelector
-     *
-     * @param {Boolean} maxParticipantsReached
-     * @returns {Array}
-     */
-    getSections(maxParticipantsReached) {
-        const sections = [];
+const NewChatPage = (props) => {
+    const [searchTerm, setSearchTerm] = useState('');
+    const [filteredRecentReports, setFilteredRecentReports] = useState([]);
+    const [filteredPersonalDetails, setFilteredPersonalDetails] = useState([]);
+    const [filteredUserToInvite, setFilteredUserToInvite] = useState();
+    const [selectedOptions, setSelectedOptions] = useState([]);
+
+    const excludedGroupEmails = _.without(CONST.EXPENSIFY_EMAILS, CONST.EMAIL.CONCIERGE);
+    const maxParticipantsReached = selectedOptions.length === CONST.REPORT.MAXIMUM_PARTICIPANTS;
+    const headerMessage = OptionsListUtils.getHeaderMessage(
+        filteredPersonalDetails.length + filteredRecentReports.length !== 0,
+        Boolean(filteredUserToInvite),
+        searchTerm,
+        maxParticipantsReached,
+    );
+    const isOptionsDataReady = ReportUtils.isReportDataReady() && OptionsListUtils.isPersonalDetailsReady(props.personalDetails);
+
+    const sections = useMemo(() => {
```

Comment:
I used `useMemo` here as it was also used here -
https://github.com/Expensify/App/blob/156d09991de1aa5f9947c8765b9142d6fe8425c1/src/pages/tasks/TaskAssigneeSelectorModal.js#L124

But I propose to remove it following the [guidelines in React docs](https://react.dev/reference/react/useMemo#should-you-add-usememo-everywhere) -
> Optimizing with useMemo  is only valuable in a few cases:
> 1. The calculation you’re putting in useMemo is noticeably slow, and its dependencies rarely change.
> 2. You pass it as a prop to a component wrapped in [memo.](https://react.dev/reference/react/memo) You want to skip re-rendering if the value hasn’t changed. Memoization lets your component re-render only when dependencies aren’t the same.
> 3. The value you’re passing is later used as a dependency of some Hook. For example, maybe another useMemo calculation value depends on it. Or maybe you are depending on this value from [useEffect.](https://react.dev/reference/react/useEffect)


2 and 3 are not applicable here. For the first point, the dependencies change often in our case which would anyway recalculate the value.

If you agree then we can remove it from `TaskAssigneeSelectorModal` too because another issue there is that it uses `props` as a dependency when it only uses props.translate during calculation -
https://github.com/Expensify/App/blob/156d09991de1aa5f9947c8765b9142d6fe8425c1/src/pages/tasks/TaskAssigneeSelectorModal.js#L163

Reply:
discussion on useCallback, useMemo in [slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1685740269693929?thread_ts=1680096510.744619&cid=C01GTK53T8Q)

Reply:
@aimane-chnaif I will remove useMemo here if you agree.